src/net/session.c

```diff
@@ -246,6 +249,7 @@ int8_t _zp_start_lease_task(_z_session_t *zn) {
                 zn->_tp._transport._multicast._lease_task_running = false;
                 ret = _Z_ERR_SYSTEM_TASK_FAILED;
                 z_free(task);
+                zn->_tp._transport._multicast._lease_task = NULL;
```

Comment:
can you also change the previous line `z_free(task)` --> `z_free(zn->_tp._transport._unicast._read_task)`.